### PR TITLE
Document asset transformation feature

### PR DESCRIPTION
**Pull request by andrewkolos**
Documents the new asset transformation feature (if curious, see https://github.com/flutter/flutter/issues/143348; but the documentation included in this PR should be sufficient for learning about and understanding this feature.)

**For reviewers.** The documentation includes a link to a yet-to-be-merged sample in the flutter/samples repo. When you arrive at this link, I ask that you review https://github.com/flutter/samples/pull/2267, which is the PR that adds the sample being linked to.

## Presubmit checklist

- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/content/ui/assets/assets-and-images.md [hunk 1]
```diff
@@ -60,6 +60,13 @@ During a build, Flutter places assets into a special
 archive called the _asset bundle_ that apps read
 from at runtime.
 
+### Automatic transformation of asset files at build time
+
+You can have asset files transformed by dart packages as they are bundled into
+your app.
+To learn how to do this and write your own asset-transforming packages, see
+[Transforming assets at build time][].
+
```
Comment (andrewkolos) [inline src/content/ui/assets/assets-and-images.md:69]: I imagine this to be a fairly niche (even if highly useful) feature, so I didn't want to fully explain it on this page but rather give it its own page and link to that. However, this section feels a bit sparse (and too easy to gloss over) to me. If anyone has ideas to improve this, please let me know.

**Review by domesticmouse — APPROVED** [review]

**Review by sfshaza2 — COMMENTED** [review]
I had this PR half reviewed yesterday. Sorry for the delay.

**Comment by sfshaza2** [comment]
Despite @domesticmouse's approval, please do not land yet.

File: src/content/ui/assets/asset-transformers.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  assets:
+    - path: assets/logo.svg
+      transformers:
+        - package: vector_graphics_compiler
+```
+
+With this configuration, `assets/logo.svg` will be transformed by the
+`vector_graphics_compiler` package as it is copied to the build output. This
+package precompiles SVG files into an optimized binary files that can be
+displayed using the `vector_graphics` package, like so:
```
[the remark this answers is not in the document]
Comment (andrewkolos) [reply src/content/ui/assets/asset-transformers.md]: See https://github.com/flutter/website/pull/10471#discussion_r1586517200

File: src/content/ui/assets/asset-transformers.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+---
+title: Transforming assets at build time
+description: How to set up automatic transformation of images (and other assets) in your Flutter app.
+short-title: Asset transformation
+---
+
+You can configure your project to automatically transform assets using
+compatible Dart packages.
```
[the remark this answers is not in the document]
Comment (andrewkolos) [reply src/content/ui/assets/asset-transformers.md]: Implemented in [make opening sentence more clear](https://github.com/flutter/website/pull/10471/commits/c9a145fe6948dcb55d57bb896c0a93b478510e70)

File: src/content/ui/assets/asset-transformers.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      transformers:
+        - package: vector_graphics_compiler
+```
+
+With this configuration, `assets/logo.svg` will be transformed by the
+`vector_graphics_compiler` package as it is copied to the build output. This
+package precompiles SVG files into an optimized binary files that can be
+displayed using the `vector_graphics` package, like so:
+
+<?code-excerpt "ui/assets_and_images/lib/logo.dart (TransformedAsset)"?>
```
[the remark this answers is not in the document]
Comment (andrewkolos) [reply src/content/ui/assets/asset-transformers.md]: Corrected a typo in my previous comment

**Comment by sfshaza2** [comment]
@andrewkolos, can you ping me directly when you feel this is ready to land?

**Comment by domesticmouse** [comment]
This should land as part of the Flutter 3.22 release with https://github.com/flutter/samples/pull/2267.

**Review by sfshaza2 — COMMENTED** [review]
We can land this when the linked to sample lands.

File: src/content/ui/assets/assets-and-images.md [hunk 5]
```diff
@@ -60,6 +60,13 @@ During a build, Flutter places assets into a special
 archive called the _asset bundle_ that apps read
 from at runtime.
 
+### Automatic transformation of asset files at build time
+
+You can have asset files transformed by dart packages as they are bundled into
+your app.
```
Comment (sfshaza2) [inline src/content/ui/assets/assets-and-images.md:66]:
```suggestion
Flutter supports using a Dart package to transform asset files when building your app.
To do this, specify the asset files and transformer package in your pubspec file.
To learn more, check out [Transforming your assets at build time][].
```

**Review by sfshaza2 — COMMENTED** [review]
A few comments. Also, can you add this new file to the sidenav? That will definitely increase visibility. Do this by adding it to /src/_data/sidenav.yml

**Comment by andrewkolos** [comment]
> few comments. Also, can you add this new file to the sidenav? That will definitely increase visibility. Do this by adding it to /src/_data/sidenav.yml

Good idea! Done.

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by andrewkolos** [comment]
Should this be reverted since 3.22 (and thus this feature) is not released yet?

**Comment by sfshaza2** [comment]
We're close enough.  :)

**Pull request merged**